src/interceptors/WebSocket/WebSocketClientConnection.ts

```diff
@@ -5,44 +5,53 @@
  * meant to be used over any WebSocket implementation
  * (not all of them follow the one from WHATWG).
  */
-import type { WebSocketRawData, WebSocketTransport } from './WebSocketTransport'
+import type { WebSocketData, WebSocketTransport } from './WebSocketTransport'
 import { WebSocketMessageListener } from './WebSocketOverride'
 import { bindEvent } from './utils/bindEvent'
 import { CloseEvent } from './utils/events'
 import { uuidv4 } from '../../utils/uuid'
 
 const kEmitter = Symbol('kEmitter')
 
+export interface WebSocketClientConnectionProtocol {
```

Comment:
Exporting this protocol means that other consumers, like MSW, can implement custom client connection-like classes. For example, to support `.clients` in the browser, MSW creates virtual connection objects that use `BroadcastChannel` to implement methods like `send()` (signal other clients from other runtimes that they should now receive data).